GitUI/ScriptsEngine/ScriptsManager.ScriptRunner.cs

```diff
@@ -55,98 +55,106 @@ private static bool RunScriptInternal(ScriptInfo script, IWin32Window owner, IGi
-                    bool success = FormProcess.ShowDialog(owner, uiCommands as GitUICommands, argument, uiCommands.GitModule.WorkingDir, null, true, process: command);
-                    if (!success)
-                    {
-                        return false;
+                            if (revisionRef is not null)
+                            {
+                                uiCommands.BrowseRepo.GoToRef(revisionRef, true);
+                            }
+                        }
+
+                        return true;
                     }
 
-                    uiCommands.RepoChangedNotifier.Notify();
-                }
-                else
-                {
-                    if (originalCommand.Equals("{openurl}", StringComparison.CurrentCultureIgnoreCase))
+                    if (!script.RunInBackground)
                     {
-                        OsShellUtil.OpenUrlInDefaultBrowser(argument);
+                        // TODO: Remove downcast from IGitUICommands to GitUICommands after https://github.com/gitextensions/gitextensions/pull/11269
+                        bool success = FormProcess.ShowDialog(owner, uiCommands as GitUICommands, argument, uiCommands.GitModule.WorkingDir, null, true, process: command);
+                        if (!success)
+                        {
+                            return false;
+                        }
+
+                        uiCommands.RepoChangedNotifier.Notify();
                     }
                     else
                     {
-                        // It is totally valid to have a command without an argument, e.g.:
-                        //    Command  : myscript.cmd
-                        //    Arguments: <blank>
-                        new Executable(command, uiCommands.GitModule.WorkingDir).Start(argument ?? string.Empty);
+                        if (originalCommand.Equals("{openurl}", StringComparison.CurrentCultureIgnoreCase))
+                        {
+                            OsShellUtil.OpenUrlInDefaultBrowser(argument);
+                        }
+                        else
+                        {
+                            // It is totally valid to have a command without an argument, e.g.:
+                            //    Command  : myscript.cmd
+                            //    Arguments: <blank>
+                            new Executable(command, uiCommands.GitModule.WorkingDir).Start(argument ?? string.Empty);
+                        }
                     }
-                }
 
-                return true;
+                    return true;
+                }
+                catch (OperationCanceledException)
+                {
+                    MessageBox.Show(owner, TranslatedStrings.ScriptUserCanceledRun, TranslatedStrings.ScriptText, MessageBoxButtons.OK, MessageBoxIcon.Information);
+                    return false;
+                }
```

Comment:
Why do we need this at all?

Reply:
Because otherwise the user have an frightening error message dialog that absolutely doesn't correspond to the reason why we abort the launch of the script:

![image](https://github.com/gitextensions/gitextensions/assets/460196/ab67a954-4437-4e96-97af-9d54f9be343a)

Reply:
To me this is a confirmation that we're doing it incorrectly :)
I think this is a misplaced responsibility, please have a look at https://github.com/gitextensions/gitextensions/commit/4b772c4729a6dc6299f9bdebaa69fe3edf7e60a0

Reply:
@RussKie Not really convinced but OK for me. Have a look at other changes required by this solution and provide feedback before I spend time to adapt tests (because with this solution, it's now less straightforward and I don't want to spend too much time on it if later adaptations are needed).

Reply:
@RussKie Friendly reminder 😉